operator/pkg/reconciliation/construct_statefulset.go

```diff
@@ -80,6 +79,29 @@ func shouldDefineSecurityContext(dc *api.CassandraDatacenter) bool {
 	return dc.Spec.ServerType == "dse" || images.CalculateDockerImageRunsAsCassandra(dc.Spec.ServerVersion)
 }
 
+func rackNodeAffinitylabels(dc *api.CassandraDatacenter, rackName string) (nodeAffinityLabels map[string]string,
+	Error error) {
+	racks := dc.GetRacks()
+	for _, rack := range racks {
+		if rack.Name == rackName {
+			nodeAffinityLabels = utils.MergeMap(emptyMapIfNil(rack.NodeAffinityLabels),
+				emptyMapIfNil(dc.Spec.NodeAffinityLabels))
+			if rack.Zone != "" {
+				if _, found := nodeAffinityLabels["zone"]; found {
+					err := fmt.Errorf("Deprecated parameter Zone is used and also defined in " +
+						"NodeAffinityLabels. You should only define it in NodeAffinityLabels")
+					return nil, err
```

Comment:
I think we can just document that the labels override the zone field, but I think logging the above message is a good idea too

Reply:
Okay so logging but not failing, correct ?

Reply:
yeah I think so